### DISCUSSION TITLE
Signal From Another Application

### DISCUSSION THREAD

**Discussion by CansuGoren** [elsa-workflows/elsa-core#1201] [Q&A]
After register documents and information control, If everything is ok, I want to resume my workflow and I thought that I could do that using send signal.
How can I send a signal from another application( like running with angular js frontend), is it possible?
Thank you in advance.

**Comment by sfmskywalker** [comment]
Certainly. There are 2 API endpoints you can invoke to send a signal from your front-end app:

- v1/signals/{signalName}/execute
- v1/signals/{signalName}/dispatch

The difference between execute and dispatch is that the former will execute the workflow before returning an HTTP response, while the latter will schedule the workflow for execution on a background queue and return an HTTP response immediately.

To send additional workflow input, workflow instance ID and/or correlation ID, take a look at the source code to see what JSON model to post with:

https://github.com/elsa-workflows/elsa-core/tree/master/src/server/Elsa.Server.Api/Endpoints/Signals

**Reply by VahidShir** [reply]
The API doesn't work when I fill WorkflowInstanceId value. This works only when the body values are null.
My question is how to trigger a signal name only associated to a specific workflow instance (in case we have multiple workflows having one identical signal name).
```
{
    "WorkflowInstanceId": "85319442683c4189867f27daf8b28162",
    "CorrelationId": null,
    "Input": null
}
```

**Reply by sfmskywalker** [reply]
I see. That sounds like a bug. Would you mind creating an issue for this with repro steps you provided here? We’ll fix it before the 2.1 release. Thanks!

**Reply by VahidShir** [reply]
Sure, I'll do. But I'm not sure I'm using it in the correct way.

**Reply by sfmskywalker** [reply]
Thank you. That’s OK, that’s why your repro will help too.

**Reply by VahidShir** [reply]
#1208
